### ISSUE TITLE
Important announcement!

### ISSUE THREAD

**Issue by citronkaste** [CodeKJ/DISIFY#13]
Hello everyone!

I would like to announce upcoming changes to API. In upcoming month or two site will be updated with new functionality and core framework will be changed.

**Don't worry, existing API structure will remain the same** and its not mandatory to update your current API requests - more data might be available in responses, but from request side you are not forced to change anything. No downtime is expected, however, please follow this thread for updates.

### New site and API version will include following changes and improvements:
- **Massive update on blacklisted MX records** [**_UPDATE_**]
Current MX blacklist ([here](https://disify.com/blacklist/mx)) contains around 450 records. However, I have checked all our currently blacklisted domains (around 40,000) and their corresponding MX records and new MX blacklist currently has more than 3,000 records. I have already removed some of false-positive ones from most popular providers (more than 100), but **I would appreciate any help** for identifying more false MX records currently listed which are actually valid and non-disposable. [Here are available new MX blacklist](https://gist.github.com/CodeKJ/543a78f8a3264ddde91240d717986f84). Please post corrections here or in Gist comments - will appreciate any contribution, thank you!

- **Improved detection of disposable emails** [**_UPDATE_**]
Emails which are not listed in our internal blacklists (domains, mx). There are often providers with healthy MX records and frequently changing domains which we can't blacklist so easily. However, some of them usually have repeated pattern on email usernames. I have implemented prediction mechanism for some of these providers, and it can be easily extended in future. Of course, this detection won't work if you just check domains instead of full email address.

- **Email username scoring** [**_NEW_**]
With this feature you will be able to see likelihood score (0 to 100) of real emails. It works by checking if email username contains real names, surnames, nicknames, years and numeration combinations. Unrecognized words and random string usernames will return very low score.

- **MX record value output** [**_NEW_**]
Responses will include MX host-names and values (if any).

- **Username output** [**_NEW_**]
Responses will include email username part (if any).

- **Malicious IP and location check** [**_NEW_**]
New endpoint will be available to check IP addresses. I have sourced database with more than 200,000 malicious IPs (updated on daily bases). If IP will be detected as malicious, API response will include amount of blacklists its been listed. You could then use this blacklist count on your side to decide on further actions if it exceeds some threshold. All responses will additionally include IP location - country, city and coordinates.

- **Account registration** [**_NEW_**]
You will have possibility to create account and use API key for some additional features, such as - analytics, statistics, request throttle increase, more data available in responses, SMTP validation (currently experimental). Even more features will be available in future for registered accounts.

Also, you might have noticed there are now available dedicated [documentation](https://docs.disify.com) page for all API requests. All newest API changes and features will be posted there.

If you have any questions or recommendations regarding updates feel free to reply. And again, I would really appreciate help for identifying false-positive MX records in [new blacklist](https://gist.github.com/CodeKJ/543a78f8a3264ddde91240d717986f84), thank you!

**Comment by Lietsaki** [comment]
This is amazing. Thank you very much for making it free!

**Comment by citronkaste** [comment]
Quick update.

New version is still in progress, but due to private reasons, last couple of months I haven't had much time to actively work and finalize it.
**ETA early 2021.**

Another feature that will be included for registered accounts are personalized whitelist/blacklist - you will have possibility to block domains which are not included in our blacklist, and allow domains which are detected by our blacklist.

Stay safe!

**Comment by DMZakaria** [comment]
@CodeKJ are you planning to share the code and make open source for contribution ?

**Comment by blackarcanis** [comment]
@CodeKJ Hi! Thank you for your work. Disify is still maintained? I mean, is the database updated regularly? 
What about new features maybe?

**Comment by citronkaste** [comment]
> @CodeKJ Hi! Thank you for your work. Disify is still maintained? I mean, is the database updated regularly? What about new features maybe?

Yes. I still maintain it. And blacklists are usually updated on monthly basis.

However, development on new version is currently on-hold.

**Comment by blackarcanis** [comment]
Hi @CodeKJ !
I hope you are well; I checked the latest blacklist update, and no new addition for more than six months.
Do you still update this and work on your new version? Don't want to bother you with this (and it's fine if you are not working on this anymore); just want to be sure to use an up-to-date service that's all ;)

**Comment by rykermorgan** [comment]
Hi @CodeKJ wondering when this update is slated for release? Also, is SMTP validation in the planned update? Right now I can submit emails I know are bad but validate because the service isn't doing mailbox validation.

**Comment by citronkaste** [comment]
Update is on-hold for indefinite period. This is one of my side-projects which I don't have much time to spend on, unfortunately. Its incomplete and not ready, and I hope to find some time to finish it, including new ideas. I haven't abandoned project, its just always other priorities.

But, I still keep updating blacklists in current version.
